pkg/controller/deployment/deployment_controller.go

```diff
@@ -72,7 +72,7 @@ type DeploymentController struct {
 	eventRecorder record.EventRecorder
 
 	// To allow injection of syncDeployment for testing.
-	syncHandler func(dKey string) error
+	syncHandler func(ctx context.Context, dKey string) error
```

Comment:
still a few `context.TODO` in the following files:
pkg/controller/deployment/progress.go
pkg/controller/deployment/util/deployment_util.go
pkg/controller/deployment/rollback.go
pkg/controller/deployment/sync.go